### PR TITLE
Only ask for `crossing:light` on tram crossings if `crossing:signals=no`

### PR DESCRIPTION
**Pull request by wielandb**
This PR splits the condition for the `crossing:light` question into seperate conditions for normal and tram crossings, with the tram crossing having the additional condition that `crossing:signals` must be `no`.

### CODE REVIEW

**Review by Robbendebiene — COMMENTED** [review]
Currently OpenStop does not write `crossing:signals` itself. It only writes to the `crossing` tag for highway crossings.

https://github.com/OPENER-next/OpenStop/blob/1f054b4afbcad80b9bb5060aa31398ee7f69ba27/assets/question_catalog/definition.json#L2684-L2732

We should introduce tram questions for the following tags like we already have for highway crossings.
```
button_operated=*
traffic_signals:sound=*
traffic_signals:vibration=*
```
@wielandb would be nice if you could extend the PR.
Check if you can reuse the existing questions by simply extending their conditions.

Additionally we need to discuss about `crossing:signals=*`.
Do we want to use it for tram crossings or do we want to use `crossing=*` like we do for highway crossings?
We will probably need a new question for traffic lights at tram crossings especially when we decide to merge it with `crossing:light`.

File: assets/question_catalog/definition.json [hunk 1]
```diff
@@ -3217,10 +3217,18 @@
...
-               "railway": ["crossing", "tram_crossing"],
+               "railway": "crossing",
                "crossing:light": false
             },
             "osm_element": "Node"
+         },
+         {
+            "osm_tags": {
+               "railway": "tram_crossing",
+               "crossing:signals": "no",
```
[the remark this answers is not in the document]
Comment (wielandb) [reply assets/question_catalog/definition.json:3228]: > Alternatively this could also be merged with the (not yet existing) crossing:signals question to a List question. The answers would be like "warning light", "traffic light", "none".
I really like this idea! I will try to implement it that way!

**Comment by wielandb** [comment]
> We should introduce tram questions for the following tags like we already have for highway crossings.

As far as I can see, all theese questions are already asked for tram_crossings. 🤔

File: assets/question_catalog/definition.json [hunk 2]
```diff
@@ -2974,6 +2974,50 @@
...
+               "image": "assets/question_catalog/images/traffic_signals_stop.png",
+               "osm_tags": {
+                  "crossing:signals": "yes",
+                  "crossing:light": "no"
+               }
+            },
+            {
+               "name": "@tramCrossingTrafficSignalWarningOption",
+               "description": "@tramCrossingTrafficSignalWarningOptionDescription",
+               "image": "assets/question_catalog/images/traffic_signals_stop.png",
```
Comment (Robbendebiene) [inline assets/question_catalog/definition.json:2997]: Wrong image path

File: assets/question_catalog/locales/en.arb [hunk 3]
```diff
@@ -468,6 +468,20 @@
   "@trafficLightVibrationSignalText" : {},
   "trafficLightVibrationSignalDescription" : "Like the acoustic signal, the vibration signal must usually be requested by pressing a button. When this button vibrates, it signals blind people that they can safely cross the street.",
   "@trafficLightVibrationSignalDescription" : {},
+  "tramCrossingTrafficSignalName" : "Tram crossing traffic signal",
+  "@tramCrossingTrafficSignalName" : {},
+  "tramCrossingTrafficSignalText" : "What kind of traffic signal is present at this tram crossing?",
```
Comment (Robbendebiene) [inline assets/question_catalog/locales/en.arb:473]: Should we rephrase this to something like this:
- "Is there a traffic signal at this tram crossing?"
- "Does the tram crossing have a traffic signal?"

Yes it is not a bool question, but having "none" as an answer makes "What kind of is present" weird to me.

**Comment by Robbendebiene** [comment]
In a recent PR keys and strings containing `traffic light` were renamed to `pedestrian lights` so we should move away from traffic signals here as well.

Since the warning light is not a real pedestrian light I kinda prefer
"Tram crossing light signal"/ `tramCrossingLightSignalName`
or
"Tram crossing signal lights"/ `tramCrossingSignalLightsName`

File: assets/question_catalog/locales/en.arb (hunk 3 above)
Comment (wielandb) [reply assets/question_catalog/locales/en.arb:473]: Would the following be okay wording?

"Is there a traffic signal at this tram crossing?"

- "Yes, a warning light"
- "Yes, a stop light"
- "No"

Comment (Robbendebiene) [reply assets/question_catalog/locales/en.arb:473]: Answers:
- Stop light
- Warning light
- No signal lights

**Comment by wielandb** [comment]
Closing this in favour of the much cleaner #239
